Formula/lgeneral.rb

```diff
@@ -0,0 +1,38 @@
+class Lgeneral < Formula
+  desc "Turn-based strategy engine heavily inspired by Panzer General"
+  homepage "https://lgames.sourceforge.io/LGeneral/"
+  url "https://downloads.sourceforge.net/lgeneral/lgeneral/lgeneral-1.4.4.tar.gz"
+  sha256 "0a26b495716cdcab63b49a294ba31649bc0abe74ce0df48276e52f4a6f323a95"
+  license "GPL-2.0-or-later"
+
+  depends_on "gettext"
+  depends_on "sdl12-compat"
+  depends_on "sdl2"
+
+  def install
+    inreplace "configure", "#include <unistd.h>", "#include <sys/stat.h>\n#include <unistd.h>"
+    system "./configure", *std_configure_args,
+                         "--disable-silent-rules",
+                         "--disable-sdltest"
+    system "make", "install"
+  end
+
+  def post_install
+    %w[nations scenarios units sounds maps gfx].each { |dir| (pkgshare/dir).mkpath }
+    %w[flags units terrain].each { |dir| (pkgshare/"gfx"/dir).mkpath }
+  end
+
+  def caveats
+    <<~EOS
+      Requires pg-data.tar.gz or the original DOS version of Panzer General. Can be downloaded from
+      https://sourceforge.net/projects/lgeneral/files/lgeneral-data/pg-data.tar.gz/download
+      To install use:
+      lgc-pg -s <pg-data-unziped-dir> -d #{share}/lgeneral
+    EOS
+  end
+
+  test do
+    system "lgeneral", "--version"
+    system "lgeneral & sleep 3; kill $!"
```

Comment:
```suggestion
    system bin/"lgeneral", "--version"
    return if OS.linux? && ENV["HOMEBREW_GITHUB_ACTIONS"].present?

    pid = fork do
      exec bin/"lgeneral"
    end
    sleep 3
    ensure
      Process.kill "TERM", pid
```
Is there a way to tell that `lgeneral` did not crash? Can we get it to generate logs which we can validate, or similar?

Reply:
Looks like there is no signal handling code(signal,sigaction) in lgeneral. So, I see no option here. 
About CI crash, https://github.com/Homebrew/homebrew-core/runs/6813150751?check_suite_focus=true#step:7:89 this probably happens because executable exited immediatly without X , pid is null and kill fails. I  beleive we need to comletly avoid running lgenerel GUI run test on Linux.

Reply:
working on it

Reply:
done